api.go

```diff
@@ -0,0 +1,1997 @@
+		Name:       opGetObject,
+		HTTPMethod: "GET",
+		HTTPPath:   "/{Bucket}/{Key+}",
+	}
+
+	if input == nil {
+		input = &s3.GetObjectInput{}
+	}
+
+	output = &GetObjectOutput{}
+	req = c.newRequest(op, input, output)
+	return
+}
+
+// GetObjectExt API operation for ECS Extension.
+func (c *S3) GetObjectExt(input *s3.GetObjectInput) (*GetObjectOutput, error) {
+	req, out := c.GetObjectExtRequest(input)
+	return out, req.Send()
+}
+
+// GetObjectExtWithContext is the same as GetObject with the addition of
+// the ability to pass a context and additional request options.
+func (c *S3) GetObjectExtWithContext(ctx aws.Context, input *s3.GetObjectInput, opts ...request.Option) (*GetObjectOutput, error) {
+	req, out := c.GetObjectExtRequest(input)
+	req.SetContext(ctx)
+	req.ApplyOptions(opts...)
+	return out, req.Send()
+}
+
+const opGetSystemMetdataSearchKeys = "GetSystemMetdataSearchKeys"
+
+// GetSystemMetdataSearchKeysRequest generates request.Request
+func (c *S3) GetSystemMetdataSearchKeysRequest(input *GetSystemMetdataSearchKeysInput) (req *request.Request, output *GetSystemMetdataSearchKeysOutput) {
+	op := &request.Operation{
+		Name:       opGetSystemMetdataSearchKeys,
+		HTTPMethod: "GET",
+		HTTPPath:   "/?searchmetadata",
+	}
+
+	if input == nil {
+		input = &GetSystemMetdataSearchKeysInput{}
+	}
+
+	output = &GetSystemMetdataSearchKeysOutput{}
+	req = c.newRequest(op, input, output)
+	return
+}
+
+// GetSystemMetdataSearchKeys API operation for ECS Extension.
+func (c *S3) GetSystemMetdataSearchKeys(input *GetSystemMetdataSearchKeysInput) (*GetSystemMetdataSearchKeysOutput, error) {
+	req, out := c.GetSystemMetdataSearchKeysRequest(input)
+	return out, req.Send()
+}
+
+// GetSystemMetdataSearchKeysWithContext is the same as GetSystemMetdataSearchKeys with the addition of
+// the ability to pass a context and additional request options.
+func (c *S3) GetSystemMetdataSearchKeysWithContext(ctx aws.Context, input *GetSystemMetdataSearchKeysInput, opts ...request.Option) (*GetSystemMetdataSearchKeysOutput, error) {
+	req, out := c.GetSystemMetdataSearchKeysRequest(input)
+	req.SetContext(ctx)
+	req.ApplyOptions(opts...)
+	return out, req.Send()
+}
+
+const opHeadBucket = "HeadBucket"
+
+// HeadBucket API operation for ECS Extension
+func (c *S3) HeadBucketExt(input *s3.HeadBucketInput) (*HeadBucketOutput, error) {
+	req, out := c.HeadBucketExtRequest(input)
+	return out, req.Send()
+}
+
+// HeadBucketExtRequest generates a request.Request
+func (c *S3) HeadBucketExtRequest(input *s3.HeadBucketInput) (req *request.Request, output *HeadBucketOutput) {
+	op := &request.Operation{
+		Name:       opHeadBucket,
+		HTTPMethod: "HEAD",
+		HTTPPath:   "/{Bucket}",
+	}
+
+	if input == nil {
+		input = &s3.HeadBucketInput{}
+	}
+
+	output = &HeadBucketOutput{}
+	req = c.newRequest(op, input, output)
+	return
+}
+
+// HeadBucketExt API operation for ECS Extension.
+func (c *S3) HeadBucketExtg(input *s3.HeadBucketInput) (*HeadBucketOutput, error) {
```

Comment:
Why "Extg"? Should that be "Ext"? It's not clear to me what this means, so use a better name if it's not.